src/puppetlabs/trapperkeeper/services/webserver/jetty9_core.clj

```diff
@@ -270,25 +273,34 @@
   (connector* server config nil))
 
 (schema/defn ^:always-validate
-  queue-thread-pool :- QueuedThreadPool
-  [max-threads :- schema/Int
-   queue-max-size :- schema/Int]
-  (let [queue-min-size (min config/default-queue-min-threads queue-max-size)]
-    (QueuedThreadPool. max-threads
-                       queue-min-size
-                       config/default-queue-idle-timeout
-                       (BlockingArrayQueue.
-                         queue-min-size
-                         queue-min-size
-                         queue-max-size))))
+  queue-thread-pool :- (schema/maybe QueuedThreadPool)
+  [max-threads :- (schema/maybe schema/Int)
+   queue-max-size :- (schema/maybe schema/Int)]
+  (if (or max-threads queue-max-size)
+    (let [thread-pool (if max-threads
+                        (QueuedThreadPool. max-threads)
+                        (QueuedThreadPool.))]
+      (if queue-max-size
+        (let [min-threads    (.getMinThreads thread-pool)
```

Comment:
Most of it is just hard-coded numbers in Jetty - see https://github.com/eclipse/jetty.project/blob/jetty-9.2.10.v20150310/jetty-util/src/main/java/org/eclipse/jetty/util/thread/QueuedThreadPool.java#L65-L98.  The only part that's somewhat more interesting is the max (_minThreads, 8) part, which the line below this in the Clojure code mimics.  The assumption being that the "8" hard-coded in two places in the Jetty code is representative of an intention to tie min-threads to the initial capacity of the thread pool's queue.

I'm not too worried about the expense of the double-construction.  Appears to mostly be setting up primitive values for fields in the `QueuedThreadPool` object.  The backing `BlockingArrayQueue` object will allocate an Object array per the initial capacity of the queue, which won't be larger than 8 - see https://github.com/eclipse/jetty.project/blob/jetty-9.2.10.v20150310/jetty-util/src/main/java/org/eclipse/jetty/util/BlockingArrayQueue.java#L113-L118.  The expensive operations wouldn't happen until after the thread pool is started, which it never would be for the first one allocated here.

I think I'd still prefer the double-construction to hard coding defaults in our code since this is a one-time thing at startup and is pretty inexpensive.